pkg/controller/identity/iamrolepolicyattachment/controller.go

```diff
@@ -125,90 +123,61 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 		}, nil
 	}
 
+	current := cr.Spec.ForProvider.DeepCopy()
+	iam.LateInitializePolicy(&cr.Spec.ForProvider, attachedPolicyObject)
+	if !reflect.DeepEqual(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+
 	cr.SetConditions(runtimev1alpha1.Available())
 
-	cr.UpdateExternalStatus(*attachedPolicyObject)
+	cr.Status.AtProvider = iam.GenerateRolePolicyObservation(*attachedPolicyObject)
 
 	return managed.ExternalObservation{
-		ResourceExists: true,
+		ResourceExists:   true,
+		ResourceUpToDate: cr.Spec.ForProvider.PolicyARN == cr.Status.AtProvider.AttachedPolicyARN,
```

Comment:
As this is not an update-able object, `ResourceUpToDate` can always return `true`.